nx/lib/nx/backend.ex

```diff
@@ -159,6 +158,7 @@ defmodule Nx.Backend do
   @callback all_close(out :: tensor, tensor, tensor, keyword) :: tensor
   @callback top_k(out :: tensor, tensor, keyword) :: tensor
   @callback take(out :: tensor, input :: tensor, indices :: tensor, keyword) :: tensor
+  @callback take_along_axis(out :: tensor, input :: tensor, indices :: tensor, axis) :: tensor
```

Comment:
```suggestion
  @callback take_along_axis(out :: tensor, input :: tensor, indices :: tensor, keyword) :: tensor
```

Reply:
On second thoughts, I'll revert this: since axis is the only option all the implementations directly accepted the axis.